### PR TITLE
Update readme

### PR DESCRIPTION
**Pull request by ncdc**
Simplify the readme:

- Show highlights first
- Simplified quickstart
- Provide links to other docs for more details

Notes for reviewer(s):

- This isn't perfect / final 😄 
- We'll iterate many times in the future
- If possible, please limit review comments to major "blockers" (e.g. something I wrote is fundamentally wrong) ❤️

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,250 +1,165 @@
...
 
-# `kcp` provides a true multi-tenant Kubernetes control plane for workloads on many clusters
+# kcp
 
-`kcp` is a generic [CustomResourceDefinition](https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/) (CRD) apiserver that is divided into multiple "[logical clusters](docs/investigations/logical-clusters.md)" that enable multitenancy of cluster-scoped resources such as CRDs and Namespaces. Each of these logical clusters is fully isolated from the others, allowing different teams, workloads, and use cases to live side by side.
+## Overview
+kcp is a Kubernetes-like control plane focusing on:
 
-<br clear="left"/>
+- A **control plane** for multiple independent, **isolated** “clusters” known as **workspaces**
```
Comment (sttts) [inline README.md:8]:
```suggestion
- A **control plane** for many independent, **isolated** “clusters” known as **workspaces**
```

File: README.md [hunk 2]
```diff
@@ -1,250 +1,165 @@
...
+- Enabling API service providers to **offer APIs centrally** using **multi-tenant operators**
+- Easy **API consumption** for users in their workspaces
+- Flexible **scheduling** of workloads to physical clusters
+- **Transparent movement** of workloads among compatible physical clusters
+- **Advanced deployment strategies** for scenarios such as affinity/anti-affinity, geographic replication, cross-cloud replication, etc.
 
-![build status badge](https://github.com/kcp-dev/kcp/actions/workflows/ci.yaml/badge.svg)
+kcp can be a building block for SaaS service providers who need a **massively multi-tenant platform** to offer services to a large number of fully isolated tenants using Kubernetes-native APIs. This could hopefully be useful to cloud providers as well as enterprise IT departments offering APIs within their company.
```
Comment (sttts) [inline README.md:15]: some other wording without the word "hopefully" maybe. "hopefully" sounds defensive, not convinced. Would formulate it more around "vision" or "goals".

Comment (sttts) [reply README.md:15]: Maybe: "The goal is to be useful ...."

Comment (ncdc) [reply README.md]: Updated, PTAL

File: README.md (hunk 1 above)
Comment (ncdc) [reply README.md]: Updated, PTAL

**Comment by sttts** [comment]
:shipit: 

/lgtm

**Pull request merged**